core/cthreads.c

```diff
@@ -0,0 +1,291 @@
+  args.func(args.data);
+
+  free(data);
+
+  return TRUE;
+}
+#endif
+
+int cthreads_thread_create(struct cthreads_thread *thread, void *(*func)(void *data), void *data) {
+  #ifdef __unix__
+    pthread_t pthread;
+
+    struct cthreads_args *args = malloc(sizeof(struct cthreads_args));
+    args->func = func;
+    args->data = data;
+
+    int res = pthread_create(&pthread, NULL, __cthreads_pthread_function_wrapper, (void *)args);
+
+    thread->pThread = pthread;
+
+    return res;
+  #elif _WIN32
+    struct cthreads_args *args = malloc(sizeof(struct cthreads_args));
+    args->func = func;
+    args->data = data;
+
+    HANDLE thread = CreateThread(NULL, 0, __cthreads_winthreads_function_wrapper, args, 0, NULL);
+
+    threading->wThread = thread;
+
+    return 0;
+  #endif
+}
+
+int cthreads_thread_detach(struct cthreads_thread *thread) {
+  #ifdef __unix__
+    return pthread_detach(thread->pThread);
+  #elif _WIN32
+    return CloseHandle(thread->wThread);
+  #endif
+}
+
+void cthreads_thread_close(void *code) {
+  #ifdef __unix__
+    pthread_exit(code);
+  #elif _WIN32
+    ExitThread((DWORD)code);
+  #endif
+
+  return;
+}
+
+struct cthreads_mutex __cthreads_mutex_init() {
+  struct cthreads_mutex mutex;
+
+  #ifdef __unix__
+    pthread_mutex_t pMutex = PTHREAD_MUTEX_INITIALIZER;
+    pthread_mutex_init(&pMutex, NULL);
+
+    mutex.pMutex = pMutex;
+  #elif _WIN32
+    HANDLE wMutex = CreateMutex(NULL, FALSE, NULL);
+
+    mutex.wMutex = wMutex;
+  #endif
+
+  return mutex;
+}
+
+int cthreads_mutex_init(struct cthreads_mutex *mutex) {
+  #ifdef __unix__
+    pthread_mutex_t pMutex = PTHREAD_MUTEX_INITIALIZER;
+    int res = pthread_mutex_init(&pMutex, NULL);
+
+    mutex->pMutex = pMutex;
+
+    return res;
+  #elif _WIN32
+    HANDLE wMutex = CreateMutex(NULL, FALSE, NULL);
+
+    if (wMutex == NULL) return 1;
+    else {
+      mutex->wMutex = wMutex;
+      return 0;
+    }
+  #endif
+}
+
+int cthreads_mutex_lock(struct cthreads_mutex *mutex) {
+  #ifdef __unix__
+    return pthread_mutex_lock(&mutex->pMutex);
+  #elif _WIN32
+    DWORD ret = WaitForSingleObject(mutex->wMutex, INFINITE);
+
+    if (ret == WAIT_OBJECT_0) return 0;
+    else return 1;
+  #endif
+}
+
+int cthreads_mutex_trylock(struct cthreads_mutex *mutex) {
+  #ifdef __unix__
+    return pthread_mutex_trylock(&mutex->pMutex);
+  #elif _WIN32
+    return WaitForSingleObject(mutex->wMutex, 0);
+  #endif
+}
+
+int cthreads_mutex_unlock(struct cthreads_mutex *mutex) {
+  #ifdef __unix__
+    return pthread_mutex_unlock(&mutex->pMutex);
+  #elif _WIN32
+    ReleaseMutex(mutex->wMutex);
+
+    return 0;
+  #endif
+}
+
+void cthreads_mutex_destroy(struct cthreads_mutex *mutex) {
+  #ifdef __unix__
+    pthread_mutex_destroy(&mutex->pMutex);
+  #elif _WIN32
+    CloseHandle(mutex->wMutex);
+  #endif
+}
+
+int cthreads_cond_init(struct cthreads_cond *cond) {
+  #ifdef __unix__
+    pthread_cond_t pCond;
+    int res = pthread_cond_init(&pCond, NULL);
```

Comment:
Would be nice if you manage to keep the attributes param.. (for both unix and windows)